cardinal/ecs/persona_integration_test.go

```diff
@@ -0,0 +1,267 @@
+
+	// Tick the game forward a bit to simulate a game that has been running for a bit of time.
+	for i := 0; i < 10; i++ {
+		assert.NilError(t, world.Tick(ctx))
+	}
+
+	_, err := world.GetSignerForPersonaTag("missing_persona", 1)
+	assert.ErrorIs(t, err, ecs.ErrPersonaTagHasNoSigner)
+
+	// Queue up a CreatePersona
+	personaTag := "foobar"
+	signerAddress := "xyzzy"
+	ecs.CreatePersonaMsg.AddToQueue(
+		world, ecs.CreatePersona{
+			PersonaTag:    personaTag,
+			SignerAddress: signerAddress,
+		},
+	)
+	// This CreatePersona will not be processed until the world.CurrentTick() is greater than the tick that
+	// originally got the CreatePersona.
+	tick := world.CurrentTick()
+	_, err = world.GetSignerForPersonaTag(personaTag, tick)
+	assert.ErrorIs(t, err, ecs.ErrCreatePersonaTxsNotProcessed)
+
+	assert.NilError(t, world.Tick(ctx))
+	// The CreatePersona has now been processed
+	addr, err := world.GetSignerForPersonaTag(personaTag, tick)
+	assert.NilError(t, err)
+	assert.Equal(t, addr, signerAddress)
+}
+
+func TestDuplicatePersonaTagsInTickAreOnlyRegisteredOnce(t *testing.T) {
+	world := testutils.NewTestWorld(t).Instance()
+	assert.NilError(t, world.LoadGameState())
+
+	personaTag := "jeff"
+
+	for i := 0; i < 10; i++ {
+		// Attempt to register many different signer addresses with the same persona tag.
+		ecs.CreatePersonaMsg.AddToQueue(
+			world, ecs.CreatePersona{
+				PersonaTag:    personaTag,
+				SignerAddress: fmt.Sprintf("address_%d", i),
+			},
+		)
+	}
+	tick := world.CurrentTick()
+
+	ctx := context.Background()
+	assert.NilError(t, world.Tick(ctx))
+
+	addr, err := world.GetSignerForPersonaTag(personaTag, tick)
+	assert.NilError(t, err)
+	// Only the first address should be associated with the user
+	assert.Equal(t, addr, "address_0")
+
+	// Attempt to register this persona tag again in a different tick. We should still maintain the original
+	// signer address.
+	ecs.CreatePersonaMsg.AddToQueue(
+		world, ecs.CreatePersona{
+			PersonaTag:    personaTag,
+			SignerAddress: "some_other_address",
+		},
+	)
+
+	assert.NilError(t, world.Tick(ctx))
+	addr, err = world.GetSignerForPersonaTag(personaTag, tick)
+	assert.NilError(t, err)
+	// The saved address should be unchanged
+	assert.Equal(t, addr, "address_0")
+}
+
+func TestCreatePersonaFailsIfTagIsInvalid(t *testing.T) {
+	// Verify that the CreatePersona is automatically created and registered with a world.
+	world := testutils.NewTestWorld(t).Instance()
+	assert.NilError(t, world.LoadGameState())
+
+	wantTag := "INVALID PERSONA TAG WITH SPACES"
+	wantAddress := "123_456"
+	ecs.CreatePersonaMsg.AddToQueue(
+		world, ecs.CreatePersona{
+			PersonaTag:    wantTag,
+			SignerAddress: wantAddress,
+		},
+	)
+	// This CreatePersona has the same persona tag, but it shouldn't be registered because
+	// it comes second.
+	ecs.CreatePersonaMsg.AddToQueue(
+		world, ecs.CreatePersona{
+			PersonaTag:    wantTag,
+			SignerAddress: "some_other_address",
+		},
+	)
```

Comment:
dont think we need this twice. we're only checking that a single message failed.

also, wantTag is a bit confusing. id suggest to remove it completely, since we won't be using it again. same with wantAddress.